coral/src/app/features/configuration/environments/components/EnvironmentStatus.tsx

```diff
@@ -0,0 +1,105 @@
+import { useToast, Box, StatusChip, Button } from "@aivenio/aquarium";
+import loading from "@aivenio/aquarium/icons/loading";
+import refresh from "@aivenio/aquarium/icons/refresh";
+import { useQuery } from "@tanstack/react-query";
+import { useState, useEffect } from "react";
+import { Environment, getUpdateEnvStatus } from "src/domain/environment";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+const EnvironmentStatus = ({
+  envId,
+  initialEnvStatus,
+}: {
+  envId: Environment["id"];
+  initialEnvStatus: Environment["envStatus"];
+}) => {
+  const toast = useToast();
+  const [shoudlRefreshStatus, setShouldRefetchStatus] = useState(false);
```

Comment:
Not something to do about, but the naming threw me of a moment, because my brain tried to make a connection to the `isRefetching` state of react-query 😬  or in general "refetching" the data from the api. Maybe a name like `updateStatus` or `getStatus` would be easier to understand.

Reply:
Oh wow yes, this is truly naming gore. `refresh` and `refetch` in the two different parts of the `useState` yikes.

Reply:
Yeah I think `shouldUpdateStatus` is better!It also makes it clear that it returns a boolean.

Reply:
[79553c1](https://github.com/Aiven-Open/klaw/pull/1761/commits/79553c16a0a78ca59e1f783dc503d35b05916317)